sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsDelete.java

```diff
@@ -15,14 +15,16 @@
  * limitations under the License.
  */
 
-package org.apache.spark.sql.sources.v2;
+package org.apache.spark.sql.connector.catalog;
 
+import org.apache.spark.annotation.Experimental;
 import org.apache.spark.sql.sources.Filter;
 
 /**
  * A mix-in interface for {@link Table} delete support. Data sources can implement this
  * interface to provide the ability to delete data from tables that matches filter expressions.
  */
+@Experimental
```

Comment:
do we also need to add the `since` tags?

Reply:
currently none of the DS v2 interfaces have the since tag. We can add them all together later.